lib/internal/streams/web.js

```diff
@@ -0,0 +1,4394 @@
+function readableByteStreamControllerCommitPullIntoDescriptor(stream, desc) {
+  assert(stream[kState].state !== 'errored');
+  let done = false;
+  if (stream[kState].state === 'closed') {
+    desc.bytesFilled = 0;
+    done = true;
+  }
+
+  const filledView =
+    readableByteStreamControllerConvertPullIntoDescriptor(desc);
+
+  if (desc.type === 'default')
+    readableStreamFulfillReadRequest(stream, filledView, done);
+  else {
+    assert(desc.type === 'byob');
+    readableStreamFulfillReadIntoRequest(stream, filledView, done);
+  }
+}
+
+function readableByteStreamControllerInvalidateBYOBRequest(controller) {
+  if (controller[kState].byobRequest === null)
+    return;
+  controller[kState].byobRequest[kState].controller = undefined;
+  controller[kState].byobRequest[kState].view = null;
+  controller[kState].byobRequest = null;
+}
+
+function readableByteStreamControllerClearAlgorithms(controller) {
+  controller[kState].pullAlgorithm = undefined;
+  controller[kState].cancelAlgorithm = undefined;
+}
+
+function readableByteStreamControllerClearPendingPullIntos(controller) {
+  readableByteStreamControllerInvalidateBYOBRequest(controller);
+  controller[kState].pendingPullIntos = [];
+}
+
+function readableByteStreamControllerGetDesiredSize(controller) {
+  const {
+    stream,
+    highWaterMark,
+    queueTotalSize,
+  } = controller[kState];
+  switch (stream[kState].state) {
+    case 'errored': return null;
+    case 'closed': return 0;
+    default: return highWaterMark - queueTotalSize;
+  }
+}
+
+function readableByteStreamControllerShouldCallPull(controller) {
+  const {
+    stream,
+  } = controller[kState];
+  if (stream[kState].state !== 'readable' ||
+      controller[kState].closeRequested ||
+      !controller[kState].started) {
+    return false;
+  }
+  if (readableStreamHasDefaultReader(stream) &&
+      readableStreamGetNumReadRequests(stream) > 0) {
+    return true;
+  }
+
+  if (readableStreamHasBYOBReader(stream) &&
+      readableStreamGetNumReadIntoRequests(stream) > 0) {
+    return true;
+  }
+
+  const desiredSize = readableByteStreamControllerGetDesiredSize(controller);
+  assert(desiredSize !== null);
+
+  return desiredSize > 0;
+}
+
+function readableByteStreamControllerHandleQueueDrain(controller) {
+  const {
+    closeRequested,
+    queueTotalSize,
+    stream,
+  } = controller[kState];
+  assert(stream[kState].state === 'readable');
+  if (!queueTotalSize && closeRequested) {
+    readableByteStreamControllerClearAlgorithms(controller);
+    readableStreamClose(stream);
+    return;
+  }
+  readableByteStreamControllerCallPullIfNeeded(controller);
+}
+
+function readableByteStreamControllerPullInto(
+  controller,
+  view,
+  readIntoRequest) {
+  const {
+    closeRequested,
+    stream,
+    pendingPullIntos,
+  } = controller[kState];
+  let elementSize = 1;
+  let ctor = DataViewCtor;
+  if (isArrayBufferView(view) && !isDataView(view)) {
+    elementSize = view.constructor.BYTES_PER_ELEMENT;
```

Comment:
this is also unreliable due to `.constructor` as well as the constructor being mutable